sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1241,30 +1270,41 @@ _pi_context::getAvailableCommandList(pi_queue Queue,
   // command lists we can create.
   // Once created, this command list & fence are added to the command list fence
   // map.
-  if (Queue->Device->Platform->ZeGlobalCommandListCount <
-      ZeMaxCommandListCacheSize) {
-    ze_command_list_handle_t ZeCommandList;
-    ze_fence_handle_t ZeFence;
 
-    uint32_t QueueGroupOrdinal;
-    auto &ZeCommandQueue =
-        Queue->getQueueGroup(UseCopyEngine).getZeQueue(&QueueGroupOrdinal);
+  // Atomically increment global count of command lists and check that we don't
+  // exceed maximum.
+  int ExpectedValue = Queue->Device->Platform->ZeGlobalCommandListCount.load();
+  int DesiredValue = 0;
+  do {
+    if (ExpectedValue >= ZeMaxCommandListCacheSize)
+      die("Exceeded the maximum amount of command lists we can create.");
+
+    DesiredValue = ExpectedValue + 1;
+  } while (!Queue->Device->Platform->ZeGlobalCommandListCount
+                .compare_exchange_strong(ExpectedValue, DesiredValue));
```

Comment:
Would we "lock" the platform object instead?
Actually, I suggest just removing the ZeGlobalCommandListCount and ZeMaxCommandListCacheSize entirely.
It doesn't make sense to have a hard limit of active command-lists

Reply:
Using compare_exchange_strong looks valid to me, it is lock-free and still thread-safe. As for me it is better than locking the platform in such popular function as getAvailableCommandList.

Ok, I will remove it then. Do you remember a reason why it was introduced?

Reply:
> Do you remember a reason why it was introduced?

Not really. What I do remember is that we needed to give people an option to bump this up. So let's just get rid of it. We don't have anything similar for other resources.

Reply:
Removed.